packages/schema/parse/src/version_validate/setUtils.ts

```diff
@@ -0,0 +1,32 @@
+export enum SetComparisionType {
+  EQUAL,
+  SUBSET,
+  DIFFERENT,
+}
+
+export function union<T>(set1: Set<T>, set2: Set<T>): Set<T> {
+  return new Set([...set1, ...set2]);
+}
+
+export function intersection<T>(set1: Set<T>, set2: Set<T>): Set<T> {
+  return new Set([...set1].filter((x) => set2.has(x)));
+}
+
+export function difference<T>(set1: Set<T>, set2: Set<T>): Set<T> {
+  return new Set([...set1].filter((x) => !set2.has(x)));
+}
+
+export function symmentric_difference<T>(set1: Set<T>, set2: Set<T>): Set<T> {
+  return difference(union(set1, set2), intersection(set1, set2));
+}
+
+export function compareSets<T>(set1: Set<T>, set2: Set<T>): SetComparisionType {
```

Comment:
In set theory, we can prove that 2 sets are equal by taking the symmetric difference of 2 sets and checking that it's an empty set. Symmetric difference is difference between union and intersection of two sets. In case of equal sets both union and intersection will be always same hence giving empty set after symmetric difference

To check if a set is a subset of another set, we can take take difference of another set with respect to the first set and if it's an empty set it proves every element in the first set are present in the second set. If none of the above cases matches it can only mean one thing that, both the set are Different.